lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -2128,23 +2110,21 @@ namespace Js
         wchar_t *outStr = builder.DangerousGetWritableBuffer();
 
         wchar_t* outStrLim = outStr + count;
+        wchar_t *o = outStr;
+
+        while (o < outStrLim)
+        {
+            *o++ = *inStr++;
+        }
 
-        //TODO the mapping based on unicode tables needs review for ES5.
-        // mapping is not always 1.1
         if(toCase == ToUpper)
         {
-            while (outStr < outStrLim)
-            {
-                *outStr++ = CharToUpperCase(*inStr++);
-            }
+            CharUpperBuffW(outStr, count);
```

Comment:
Certain characters increase the length of the string when converted to uppercase, e.g. `LATIN SMALL LETTER N PRECEDED BY APOSTROPHE`, as per [SpecialCasing.txt](http://www.unicode.org/Public/5.1.0/ucd/SpecialCasing.txt). If this API handles those characters, this call wouldn't convert some characters at the end. If not, we'll need to have a workaround for them.